### PR TITLE
[Bug 16210] Setting the styledText wipes out listStyle

### PR DESCRIPTION
**Pull request by livecodepanos**

### CODE REVIEW

**Comment by runrevmark** [comment]
@livecode-panos: There's an issue here with the textAlign too... I think there might be some machinery which could be used here to simplify the code - the textALign and listStyle properties should have appropriate enum definitions.

**Comment by trevordevore** [comment]
Would it be possible to get this fixed squared away soon? I'm guessing this is not going into LC 8 dp-8 and it is keeping me from releasing an update to software based on the 7/8 engines.

**Comment by runrevmark** [comment]
@livecodepanos: Ah! Yes - I knew there was logic elsewhere for this - check the line comments for a further improvement suggestion.

File: engine/src/paragrafattr.cpp [hunk 1]
```diff
@@ -100,11 +100,30 @@ void MCParagraph::fetchattrs(MCArrayRef src)
...
 
     // AL-2014-09-30: [[ Bug 13559 ]] Don't explicitly set any paragraph styles that aren't in the array (even to empty)
-    if (ctxt . CopyElementAsInteger(src, MCNAME("textAlign"), false, t_intenum_value))
-        SetTextAlign(ctxt, &t_intenum_value);
-
-    if (ctxt . CopyElementAsInteger(src, MCNAME("listStyle"), false, t_intenum_value))
-        SetListStyle(ctxt, t_intenum_value);
+	// PM-2015-10-14: [[ Bug 16210 ]] Get the stringref for textAlign from the src array and then map it to the correct
+	// intenum_t, to be used in SetTextAlign()
+	if (ctxt . CopyElementAsString(src, MCNAME("textAlign"), false, t_stringref_value))
```
Comment (runrevmark) [inline engine/src/paragrafattr.cpp]: This seems perfect for a MCExecContext::CopyElementAsEnum(array, keyname, casesensitive, enumtypeinfo, r_intenum) abstraction - which can be used again below too.

File: tests/lcs/core/field/listStyle.livecodescript [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+on TestStyledTextResetsListStyle
+
+open this stack
+create fld "myFld"
```
Comment (runrevmark) [inline tests/lcs/core/field/listStyle.livecodescript:22]: I'd suggest doing these tests as follows:

```
put "This is some text & return & this is some more text" into fld "myFld"
repeat for each item tListStyle in <list of styles>
  set the listStyle of char 1 to -1 of fld "myFld" to tListStyle
  put the styledText of fld "myFld" into tStyledText
  set the styledText of fld "myFld" to tStyledText

  TestAssert "setting styledText respects listStyle for '" & tListStyle & "'", the listStyle of char 1 to -1 of fld "myFld" is tListStyle
end repeat
```

File: tests/lcs/core/field/textAlign.livecodescript [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+on TestStyledTextResetsTextAlign
+
+open this stack
+create fld "myFld"
+put "This is some text & return & this is some more text" into fld "myFld"
+set the textAlign of char 1 to -1 of fld "myFld" to "center"
```
Comment (runrevmark) [inline tests/lcs/core/field/textAlign.livecodescript]: Use the same pattern as above to check each possible align value (including empty).

**Comment by livecodepanos** [comment]
Arghh! New tests for textAlign kept failing because I wrote "centre" instead of "center"!

**Comment by peter-b** [comment]
@livecodepanos Congratulations on your excellent English spelling skills!

**Comment by trevordevore** [comment]
@peter-b Or maybe he is just really good at French? ;-)

**Comment by peter-b** [comment]
@livecode-vulcan review ok 6236b46

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok 6236b46df5d3ee1b1ef9bdf8adb2b90a86e04327

**Comment by livecode-vulcan** [comment]
😎 test success 6236b46df5d3ee1b1ef9bdf8adb2b90a86e04327
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/941)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/949)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/938)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/942)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/951)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/960)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/955)

**Pull request merged**